File: cpp/decorator/StarbuzzCoffee/decorator/condiments/Mocha.cpp
#include "../CondimentDecorator.h"

class Mocha : public CondimentDecorator {

public:
    explicit Mocha(Beverage* beverage) { this->beverage = beverage; }

    string getDescription() override { return beverage->getDescription() + ", Mocha"; }

    double cost() override { return beverage->cost() + 0.99; }

private:
    Beverage* beverage;

};

File: ruby/decorator/StarbuzzCoffee/decorator/condiments/mocha.rb
require './decorator/condiment_decorator'

class Mocha < CondimentDecorator

  def initialize(beverage)
    @@beverage = beverage
  end

  def get_description
    super
    @@beverage.get_description + ", Mocha"
  end

  def cost
    super
    @@beverage.cost + 0.99
  end

  private
  @@beverage = Beverage.new

end